core/src/main/resources/lib/audio.fjs

```diff
@@ -0,0 +1,177 @@
+                , audioType: String
+                , playPauseButton: HTMLElement
+                , timePlayed: HTMLElement
+                , timeDuration: HTMLElement
+                , scrubber: HTMLElement
+                };
+type Audio = AudioF & Atom;
+
+function setPlayingState(dom: DomService, playPauseButton: HTMLElement) {
+  dom.write(() => {
+    playPauseButton.classList.add('is-playing');
+  });
+}
+
+function setPausedState(dom: DomService, playPauseButton: HTMLElement) {
+  console.log('PAUSED!');
+  dom.write(() => {
+    playPauseButton.classList.remove(['is-playing']);
+  });
+}
+
+function formatTime(t: number): string {
+  const format = (t: number) => t.toFixed(0).padStart(2, '0');
+  const second = Math.floor(t % 60);
+  const minute = Math.floor((t % 3600) / 60);
+  const hour = Math.floor(t / 3600);
+  return `${format(hour)}:${format(minute)}:${format(second)}`;
+}
+
+function setupTime(dom: DomService, audio: Audio) {
+  dom.write(() => {
+    audio.scrubber.value = 0;
+    audio.timePlayed.innerText = formatTime(0);
+    audio.timeDuration.innerText = formatTime(audio.player.duration);
+  });
+}
+
+function updateTime(dom: DomService, audio: Audio) {
+  const now = Date.now();
+  const last = audio.player.getAttribute('last-tick');
+
+  // throttle to 1 update per second
+  if (now - (Number(last) || 0) < 1000) return;
+
+  const played = Math.floor(audio.player.currentTime);
+  const percentPlayed = Math.floor(played / audio.player.duration * 100);
+  const gradientDescription = `linear-gradient(to right, #C70000 ${percentPlayed}%, #afafaf ${percentPlayed}%)`;
+
+  dom.write(() => {
+    audio.scrubber.value = percentPlayed;
+    audio.player.setAttribute('last-tick', now.toString());
+    audio.scrubber.style.background = gradientDescription;
+    audio.timePlayed.innerText = formatTime(played);
+  });
+}
+
+export default (componentType: ComponentType) => ({ ophan, dom, viewport }: Services): AtomBuilder<AudioF> => (root: HTMLElement): Coeval<Audio> => {
+  let playPauseC: Channel<Action>;
+  let scrubC: Channel<Action>;
+  let playerTimeUpdateC: Channel<Action>;
+  let observer: (x: number) => void;
+  let audioContainerSelector = '.atom--audio';
+  let audioPlayerSelector = '.atom--audio__player-element';
+  let playPauseButtonSelector = '.atom--audio__button-playaudio';
+  let progressBarSelector = '.atom--audio__progress-bar';
+  let progressSlider = '.atom--audio__progress-bar > input';
+  let timePlayedSelector = '.atom--audio__time-played > span';
+  let timeDurationSelector = '.atom--audio__time-duration > span';
+
+  const start = (a: Audio): Promise<void> => {
+    playPauseC = fromEvent('click', a.playPauseButton)
+      .map((e: Event) => ((e.target: any).closest(playPauseButtonSelector): ?HTMLButtonElement))
+      .filter((e: ?HTMLButtonElement) => !!e);
+    playPauseC.tap(onPlayPause(a));
+
+    //TODO: Do we have to capture scrub events?
+    scrubC = fromEvent('click', a.scrubber)
```

Comment:
Something for the future - but we don't need to track scrub events as such. What Stu wants is: 25% played, 50%, 75% played and completed events. Which you should be able to drive off the `updateTime` function.